### PR TITLE
DOC: Added an example for bipolar referencing

### LINKED ISSUE #8993: set_bipolar_reference is not present in the doc or any example
we should add

https://mne.tools/stable/generated/mne.set_bipolar_reference.html

to the some example and documentation.

### PR DESCRIPTION
**Pull request by ramkpari**
#### Reference issue
Fixes #8993 .

#### What does this implement/fix?
Added an example for bipolar referencing using `set_bipolar_reference` function. 

#### Additional information
 Referencing this comment (https://github.com/mne-tools/mne-python/issues/8993#issuecomment-790688368), Option 1 was implemented with a new section added below it. 

Also added a reference regarding contralateral bipolar referencing to add more context to the usage of the example being described.

### CODE REVIEW

**Comment by ramkpari** [comment]
PS: I've never used bipolar referencing till now, So it would be useful to verify if the example I added is realistic.

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 1]
```diff
@@ -76,6 +76,9 @@
 # use average of mastoid channels as reference
 # raw.set_eeg_reference(ref_channels=['M1', 'M2'])
 
+# use a Bipolar reference (contralateral)
+# raw.set_bipolar_reference(anode='[F3'],cathode=['F4'])
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:80]:
```suggestion
# raw.set_bipolar_reference(anode='[F3'], cathode=['F4'])
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 2]
```diff
@@ -188,6 +191,24 @@
...
 
+###############################################################################
+# Using a Bipolar reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a Bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
+# along the respective channel names for ``anode`` and ``cathode`` which
+# creates a new virtual channel that takes the difference between two
+# specified channels (Anode and Cathode) and drops the original channels by
+# default. The new virtual channel will be annoted with the channel info of
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:202]:
```suggestion
# default. The new virtual channel will be annotated with the channel info of
```
I assume this should say "annotated"?

Comment (ramkpari) [reply tutorials/preprocessing/plot_55_setting_eeg_reference.py]: Yeah, it should

File: doc/references.bib [hunk 3]
```diff
@@ -2049,3 +2049,15 @@ @Article{Kappenman2021
 }
 
 @Comment{jabref-meta: databaseType:bibtex;}
+
+@article{Yao2019,
```
Comment (britta-wstnr) [inline doc/references.bib:2053]: For consistency, you could rename to `YaoEtAl2019` (compare to other references).

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 4]
```diff
@@ -188,6 +191,24 @@
...
+###############################################################################
+# Using a Bipolar reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a Bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
+# along the respective channel names for ``anode`` and ``cathode`` which
+# creates a new virtual channel that takes the difference between two
+# specified channels (Anode and Cathode) and drops the original channels by
+# default. The new virtual channel will be annotated with the channel info of
+# the anode with locations set to (0,0,0) and coil types set to EEG_BIPOLAR by
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:203]:
```suggestion
# the anode with locations set to (0, 0, 0) and coil types set to EEG_BIPOLAR by
```

Comment (britta-wstnr) [reply tutorials/preprocessing/plot_55_setting_eeg_reference.py:203]: Nitpicking ;-)

File: doc/references.bib (hunk 3 above)
Comment (ramkpari) [reply doc/references.bib]: Should be updated now!

**Comment by ramkpari** [comment]
https://26138-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/preprocessing/plot_55_setting_eeg_reference.html?highlight=bipolar%20ref

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 5]
```diff
@@ -188,6 +191,24 @@
     fig.subplots_adjust(top=0.9)
     fig.suptitle('{} reference'.format(title), size='xx-large', weight='bold')
 
+###############################################################################
+# Using a Bipolar reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a Bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
+# along the respective channel names for ``anode`` and ``cathode`` which
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:199]:
```suggestion
# along with the respective channel names for ``anode`` and ``cathode`` which
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 6]
```diff
@@ -188,6 +191,24 @@
...
+# creates a new virtual channel that takes the difference between two
+# specified channels (Anode and Cathode) and drops the original channels by
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:201]:
```suggestion
# specified channels (anode and cathode) and drops the original channels by
```
Not sure, but I would not capitalize them here.

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 7]
```diff
@@ -188,6 +191,24 @@
...
+# default. The new virtual channel will be annotated with the channel info of
+# the anode with locations set to (0, 0, 0) and coil types set to EEG_BIPOLAR
+# by default. Here we use a contralateral/transverse bipolar reference
+# between channels EEG 054 and EEG 055 as described in :footcite:`YaoEtAl2019`
+# which creates a new virtual channel named EEG 054-EEG 055.
```
Comment (britta-wstnr) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:206]: Can you mark `EEG_BIPOLAR`, `EEG 054`, etc here as code? See the rest of the tutorial for reference.

**Review by britta-wstnr — COMMENTED** [review]
Output looks already good, thanks for posting the link!
I have a few small things to adjust.

I was hoping to see this somewhere at the beginning in the tutorial but didn't: where are the two channels (`EEG 054`, `EEG 055`) on the scalp? Trying to see if a bipolar ref between the two makes sense :-)

**Comment by ramkpari** [comment]
> Output looks already good, thanks for posting the link!
> I have a few small things to adjust.
> 
> I was hoping to see this somewhere at the beginning in the tutorial but didn't: where are the two channels (`EEG 054`, `EEG 055`) on the scalp? Trying to see if a bipolar ref between the two makes sense :-)

Yeah, I was worried about that as well.  Attached a sensor map with locations. 

Sensor_positions_(eeg)

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 8]
```diff
@@ -76,6 +76,9 @@
 # use average of mastoid channels as reference
 # raw.set_eeg_reference(ref_channels=['M1', 'M2'])
 
+# use a Bipolar reference (contralateral)
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:79]:
```suggestion
# use a bipolar reference (contralateral)
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 9]
```diff
@@ -188,6 +191,25 @@
     fig.subplots_adjust(top=0.9)
     fig.suptitle('{} reference'.format(title), size='xx-large', weight='bold')
 
+###############################################################################
+# Using a Bipolar reference
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:195]:
```suggestion
# Using a bipolar reference
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 10]
```diff
@@ -188,6 +191,25 @@
...
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a Bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:198]:
```suggestion
# To create a bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 11]
```diff
@@ -188,6 +191,25 @@
...
+# along with the respective channel names for ``anode`` and ``cathode`` which
+# creates a new virtual channel that takes the difference between two
+# specified channels (anode and cathode) and drops the original channels by
+# default. The new virtual channel will be annotated with the channel info of
+# the anode with locations set to (0, 0, 0) and coil types set to
+# ``EEG_BIPOLAR``by default. Here we use a contralateral/transverse bipolar
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:204]:
```suggestion
# ``EEG_BIPOLAR`` by default. Here we use a contralateral/transverse bipolar
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 12]
```diff
@@ -188,6 +191,25 @@
...
+###############################################################################
+# Using a Bipolar reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a Bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
+# along with the respective channel names for ``anode`` and ``cathode`` which
+# creates a new virtual channel that takes the difference between two
+# specified channels (anode and cathode) and drops the original channels by
+# default. The new virtual channel will be annotated with the channel info of
+# the anode with locations set to (0, 0, 0) and coil types set to
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:203]:
```suggestion
# the anode with locations set to ``(0, 0, 0)`` and coil types set to
```

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 13]
```diff
@@ -188,6 +191,25 @@
...
+# ``EEG_BIPOLAR``by default. Here we use a contralateral/transverse bipolar
+# reference between channels ``EEG 054`` and ``EEG 055`` as described in
+# :footcite:`YaoEtAl2019` which creates a new virtual channel
+# named ``EEG 054-EEG 055``.
+
+raw_bip_ref = mne.set_bipolar_reference(raw.copy(), anode=['EEG 054'],
+                                        cathode=['EEG 055'])
```
Comment (hoechenberger) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:210]: `set_bipolar_reference` created a copy by default (see its `copy` param)
```suggestion
raw_bip_ref = mne.set_bipolar_reference(raw, anode=['EEG 054'],
                                        cathode=['EEG 055'])
```

**Comment by hoechenberger** [comment]
It appears to me the doc build is failing not because of your changes …

**Comment by hoechenberger** [comment]
@ramkpari Could you fetch & merge `main` and push again please?

**Comment by ramkpari** [comment]
Yeah , I pulled the last commit and local make seems to work just fine.  It's hitting at this for some reason. 

```
changes/latest.inc:48: WARNING: py:func reference target not found: mne.epochs.make_fixed_length_epochs
```

**Comment by ramkpari** [comment]
Alright let me try that

File: tutorials/preprocessing/plot_55_setting_eeg_reference.py [hunk 14]
```diff
@@ -188,6 +191,25 @@
...
+# Using a bipolar reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To create a bipolar reference, you can use :meth:`~mne.set_bipolar_reference`
+# along with the respective channel names for ``anode`` and ``cathode`` which
+# creates a new virtual channel that takes the difference between two
+# specified channels (anode and cathode) and drops the original channels by
+# default. The new virtual channel will be annotated with the channel info of
+# the anode with locations set to ``(0, 0, 0)`` and coil types set to
+# ``EEG_BIPOLAR`` by default. Here we use a contralateral/transverse bipolar
```
Comment (drammock) [inline tutorials/preprocessing/plot_55_setting_eeg_reference.py:204]:
```suggestion
# the anode with location set to ``(0, 0, 0)`` and coil type set to
# ``EEG_BIPOLAR`` by default. Here we use a contralateral/transverse bipolar
```

**Review by drammock — APPROVED** [review]
The CircleCI error is unrelated and I'll fix it in a separate PR.  I'm making one small change and then merging. Thanks @ramkpari!

**Comment by drammock** [comment]
looks like @larsoner already fixed that in d8fc038f8

**Pull request merged**